polkadot/runtime/parachains/src/assigner_coretime/mod.rs

```diff
@@ -284,12 +284,10 @@ impl<T: Config> AssignmentProvider<BlockNumberFor<T>> for Pallet<T> {
 		})
 	}
 
-	fn report_processed(assignment: Assignment) {
-		match assignment {
-			Assignment::Pool { para_id, core_index } =>
-				on_demand::Pallet::<T>::report_processed(para_id, core_index),
-			Assignment::Bulk(_) => {},
-		}
+	fn report_processed(para_id: ParaId, core_index: CoreIndex) {
+		// Reporting processed assignments is only important for on-demand.
+		// Doing the call below is a no-op if the assignment was a `Bulk` one.
+		on_demand::Pallet::<T>::report_processed(para_id, core_index);
```

Comment:
@eskimor this is a bit of a hack but is IMO safe to do.
This helps when reporting as processed the cores that remain occupied when a session change occurs. As you suggested, we no longer back anything if we know that at the end of the block a session change will occur. Cores that remain occupied at the session change will be freed and their assignments will no longer be pushed back to the assigner, BUT we still want to report them as processed, right?

AFAICT this is only relevant for on-demand cores, but because I removed the AV-core storage, we no longer know which assignment type this was (on demand or bulk). I prefer not storing this info in the inclusion module or somewhere else so this hack seems to do the trick:
`on_demand::report_processed` seems to not do anything if the paraid had no affinity on the core.

This is a hack definitely. But it's one I'm willing to do to not add extra complexity and we'll fix this for good when removing the claim queue: https://github.com/paritytech/polkadot-sdk/issues/5529
Sounds ok?

Reply:
Yes, please no distinction between on-demand or bulk. The whole goal of the assignment provider was to hide this difference, so the rest of the code does not have to bother. 

Just report processed, if dropped. That's not even a hack. It has been processed, just not successfully. We can add this to the docs of the call, that "processed" does not necessarily mean success, but more that it had its chance and execution time.